### PR TITLE
ZEN-31635: Added snmpLastCollection to indexed device fields

### PR DESCRIPTION
**Pull request by pjohnston**

### CODE REVIEW

**Comment by dougsyer** [comment]
Not sure if this has been fixed but we have found last model time will update id some plugins fail and sometimes when they all fail.

File: Products/ZenModel/migrate/addSnmpLastCollectionIndex.py [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+
+from Products.Zuul.catalog.model_catalog_init import reindex_model_catalog
+from Products.Zuul.catalog.indexable import DeviceIndexable
+from Products.Zuul.catalog.interfaces import IModelCatalogTool
+
+log = logging.getLogger("zen.migrate")
+
+class AddSnmpLastCollectionIndex(Migrate.Step):
+
+    version = Migrate.Version(300, 0, 13)
```
Comment (jpeacock-zenoss) [inline Products/ZenModel/migrate/addSnmpLastCollectionIndex.py]: This should be
```
    version = Migrate.Version(SCHEMA_MAJOR, SCHEMA_MINOR, SCHEMA_REVISION)
```
and the import is:
```
from Products.ZenModel.ZMigrateVersion import SCHEMA_MAJOR, SCHEMA_MINOR, SCHEMA_REVISION
```

File: Products/Zuul/catalog/indexable.py [hunk 2]
```diff
@@ -375,7 +376,10 @@ def idx_hwManufacturer(self):
     def idx_serialNumber(self):
         return self.hw.serialNumber
 
-
+    @indexed(StringFieldType(indexed=True, stored=True), attr_query_name="snmpLastCollection")
```
Comment (jpeacock-zenoss) [inline Products/Zuul/catalog/indexable.py]: Any reason to not index the timestamp as a `FloatFieldType`? Timestamps are generally handled as numerical values, not strings.

File: Products/ZenModel/migrate/addSnmpLastCollectionIndex.py (hunk 1 above)
Comment (pjohnston) [reply Products/ZenModel/migrate/addSnmpLastCollectionIndex.py]: Done

File: Products/Zuul/catalog/indexable.py (hunk 2 above)
Comment (pjohnston) [reply Products/Zuul/catalog/indexable.py]: As we discussed, added as a DoubleFieldType to avoid potential precision loss.

**Pull request merged**